### PR TITLE
 On branch model-specificiation

### LINKED ISSUE #3: Model Specification + Serialization
Goal: Create class for GAN model specification. Includes: forward pass, load/save from disk. Nice to have: Make structure configurable based on the inputs we have. 

Design Inspiration: https://github.com/Akettle44/LLMs-What-Can-Pruning-Tell-Us/tree/main/src/model
Can use the above as a baseline for what is needed in terms of functionality. 

Acceptance Criteria: 
Implement all things mentioned in goal. Have unit tests verifying behavior of model and serialization (load / saving).

### PR DESCRIPTION
**Pull request by FeilongHou**
Your branch is up to date with 'origin/model-specificiation'.

 Changes to be committed:
	modified:   src/main.py
	new file:   src/model/model.py
	new file:   src/train/train.py
Commit message: Add model structure, save model parameter
commit by Feilong Hou 11/20/24

### CODE REVIEW

File: src/model/model.py [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+        # k3n128s1
+        self.block3 = DisBlock(64, 2, 1)
+        # k3n128s2
+        self.block4 = DisBlock(128, 1, 2)
+        # k3n256s1
+        self.block5 = DisBlock(128, 2, 1)
+        # k3n256s2
+        self.block6 = DisBlock(256, 1, 2)
+        # k3n512s1
+        self.block6 = DisBlock(256, 2, 1)
```
Comment (Akettle44) [inline src/model/model.py]: self.block6 is defined twice with different shapes

File: src/model/model.py [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+        x = self.block1(x)
+        output = self.block2(x)
+        output = self.block3(output)
+        output = self.block4(output)
+        output = self.block5(output)
+        output = self.block6(output)
+        output = self.block7(output)
+        output = self.block8(output)
+        output = self.block9(output)
+        output = self.block10(output + x)
```
Comment (Akettle44) [inline src/model/model.py:97]: Can you clarify that adding x here was intentional?

File: src/model/model.py (hunk 1 above)
Comment (FeilongHou) [reply src/model/model.py]: will fix that

File: src/model/model.py (hunk 2 above)
Comment (FeilongHou) [reply src/model/model.py:97]: On the paper, there is a skip connection from before residual blocks to the first block after residual blocks

**Pull request merged**